### PR TITLE
Filename width fix

### PR DESCRIPTION
**Pull request by NHOrus**
Some people have unicode filenames. They obviously misalign with naive width calculations.

### CODE REVIEW

**Comment by NHOrus** [comment]
And I am done for today.

File: tailer.go [hunk 1]
```diff
@@ -83,8 +84,9 @@ func maximumNameLength(filenames []string) int {
 	max := 0
 	for _, name := range filenames {
 		base := filepath.Base(name)
-		if len(base) > max {
-			max = len(base)
+		width := runewidth.StringWidth(base)
```
Comment (NHOrus) [inline tailer.go:87]: As this is a bit more costly than len(string), calling only once per loop and instead using temporary variable.

**Comment by gongo** [comment]
In my environment:

image

**Comment by NHOrus** [comment]
screenshot at 2017-02-04 03-22-01
Without filename width patch and with it.
Need to think more about it.

**Comment by NHOrus** [comment]
screenshot at 2017-02-04 03-27-01
And with utf8. RuneCountInString() instead of go-runewidth.

**Comment by NHOrus** [comment]
Sorry. I don't know.

**Comment by gongo** [comment]
I will try to check that out too 💪

**Comment by gongo** [comment]
Hi @NHOrus .
I came up with a good way. I might develop it on another branch. 🙏

**Comment by NHOrus** [comment]
I think if you succeed it would be great to put in it's own library. Because, well, not a question of width of symbol, but width of a glyph in font.

Also, willing to help with that.

**Comment by gongo** [comment]
> Because, well, not a question of width of symbol, but width of a glyph in font.

Sure.

But, even if it can not be solved in **all environments** , it may be able to solve to **some extent** (for example: limited to monospaced fonts) 💪

**Comment by emersion** [comment]
Maybe this can help? https://godoc.org/golang.org/x/text/width

**Comment by gongo** [comment]
@NHOrus 

I fixed it in GH-7, so close this PR. Thanks !!

**Pull request closed without merging**